### PR TITLE
Spring Cleaning IS-4644 Enable use of auto-merging resolver

### PR DESCRIPTION
**Pull request by newzac**
This should resolve issue #52.

Rollback Plan
-------------
**If this pull request requires anything more complex (e.g., rolling back a migration), you MUST update this section. Otherwise, delete this note.**

To roll back this change, revert the merge with `git revert -m 1 MERGE_SHA` and perform another deploy.

URLs
----
* [Jira issue IS-4644](https://sportngin.atlassian.net/browse/IS-4644)

QA Plan
-------
Provide a detailed QA plan, or other developers will retain the right to mock you mercilessly.

### CODE REVIEW

File: lib/octopolo/git.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Git
     STAGING_PREFIX = "staging"
     QAREADY_PREFIX = "qaready"
 
+    RESOVLER_USED = nil
```
Comment (anfleene) [inline lib/octopolo/git.rb]: Don't use  a constant for this. Make it an instance variable or something

File: lib/octopolo/pull_request_merger.rb [hunk 2]
```diff
@@ -47,7 +48,13 @@ def perform
       when GitHub::PullRequest::NotFound
         cli.say "Unable to find pull request #{pull_request_id}. Please retry with a valid ID."
       when Git::MergeFailed
-        cli.say "Merge failed. Please identify the source of this merge conflict resolve this conflict in your pull request's branch. NOTE: Merge conflicts resolved in the #{branch_type} branch are NOT used when deploying."
+        if @merge_resolver
+          %x(#{Octopolo.Config.merge_resolver})
```
Comment (anfleene) [inline lib/octopolo/pull_request_merger.rb]: Do we need this? I thought Git was covering it?

Comment (newzac) [inline lib/octopolo/pull_request_merger.rb]: You're right. I was a little confused with how it was referenced.

**Comment by newzac** [comment]
:octocat: Merged into staging.2014.07.28.

**Comment by newzac** [comment]
:octocat: Merged into staging.2014.07.28.

File: lib/octopolo/config.rb [hunk 3]
```diff
@@ -44,6 +44,10 @@ def github_repo
       @github_repo || raise(MissingRequiredAttribute, "GitHub Repo is required")
     end
 
+    def merge_resolver
+      @merge_resolver || nil
```
Comment (zserre) [inline lib/octopolo/config.rb]: why do we need an || nil here, if it is not defined it will just return nil

**Comment by carlallen** [comment]
QA looks good

File: spec/octopolo/git_spec.rb [hunk 4]
```diff
@@ -197,6 +197,7 @@ module Octopolo
         Git.should_receive(:fetch)
         Git.should_receive(:perform).with("merge --no-ff origin/#{branch_name}", :ignore_non_zero => true)
         Git.should_receive(:clean?) { true }
+        Git.should_receive(:clean?) { true }
```
Comment (zserre) [inline spec/octopolo/git_spec.rb]: why is this duplicated from the line above?

File: spec/octopolo/git_spec.rb [hunk 5]
```diff
@@ -207,6 +208,7 @@ module Octopolo
         Git.should_receive(:fetch)
         Git.should_receive(:perform).with("merge --no-ff origin/#{branch_name}", :ignore_non_zero => true)
         Git.should_receive(:clean?) { false }
+        Git.should_receive(:clean?) { false }
```
Comment (zserre) [inline spec/octopolo/git_spec.rb]: same comment here

File: spec/octopolo/git_spec.rb (hunk 4 above)
Comment (carlallen) [inline spec/octopolo/git_spec.rb]: expect(...).to receive(...).twice

Comment (carlallen) [inline spec/octopolo/git_spec.rb]: obj.should_receive(:message).once
obj.should_receive(:message).twice
obj.should_receive(:message).exactly(3).times

obj.should_receive(:message).at_least(:once)
obj.should_receive(:message).at_least(:twice)
obj.should_receive(:message).at_least(n).times

Comment (anfleene) [inline spec/octopolo/git_spec.rb]: what Carl said

**Comment by anfleene** [comment]
Let's add it to the readme too 🍻

**Comment by carlallen** [comment]
QA still good, CR good

**Comment by anfleene** [comment]
:shipit:

**Comment by newzac** [comment]
:octocat: Merged into deployable.2016.05.16.

**Pull request merged**